### PR TITLE
GitHub event collector: return last run when no events are fetched

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-25095

## Description
If there are no events collected, we don't want to change the `lastRun` object.

### CODE REVIEW

File: Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py [hunk 1]
```diff
@@ -117,7 +116,9 @@ def main():
 
             if command == 'fetch-events':
                 send_events_to_xsiam(events, vendor=VENDOR, product=PRODUCT)
-                demisto.setLastRun(GithubGetEvents.get_last_run(events))
+                if events:
```
Comment (dorschw) [inline Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py:119]: please add a unit-test

Comment (ilaner) [reply Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py:119]: There are no unit tests for this integration at all. We need to have another task to write unit tests for this collector.

Comment (ilaner) [reply Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py]: added unit test for last run

**Review by yuvalbenshalom — APPROVED** [review]

**Comment by ilaner** [comment]
Force merge is needed here since there were not tests and the new test coveres below 75 percent of the integration

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**